Docs improvements for  wait feature. (#2354)
hashicorp/terraform-provider-kubernetes: website/docs/r/manifest.html.markdown

- `rollout` (Optional) When set to `true` will wait for the resource to roll out, equivalent to `kubectl rollout status`. 
- `condition` (Optional) A set of condition to wait for. You can specify multiple `condition` blocks and it will wait for all of them. 
- `fields` (Optional) A map of field paths and a corresponding regular expression with a pattern to wait for. The provider will wait until the field's value matches the regular expression. Use `*` for any value.

> A field path is a string describing the fully qualified address of a field within the resource, including its parent fields all the way up to "object". The syntax of a path string follows the rules below:

Reviewer: ```suggestion
A field path is a string that describes the fully qualified address of a field within the resource, including its parent fields all the way up to "object". The syntax of a path string follows the rules below:
```

  * Fields of objects are addressed with `.`
  * Keys of a map field are addressed with `["<key-string>"]`
  * Elements of a list or tuple field are addresed with `[<index-numeral>]`

>   For example, to wait for a ServiceAccount token to be created in a Secret, where the `data` field of Secret is a map, do the following:

Reviewer: ```suggestion
  The following example waits for Kubernetes to create a ServiceAccount token in a Secret, where the `data` field of the Secret is a map.
```

  ```hcl
  wait {
    fields = {
      "data[\"token\"]" = ".+"
    }
  }
  ```

>   If unsure which type a certain field is, the [`type()`]() Terraform function can determine it. With the resource created and present in state, run `terraform console` and then the following command:

Reviewer: ```suggestion
  You can use the [`type()`]() Terraform function to determine the type of a field. With the resource created and present in state, run `terraform console` and then the following command:
```
